---
title: "Tara_analysis"
output:
  pdf_document: default
  html_document: default
---

# TARA Oceans Data Analysis

## Setup and Package Loading

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo= TRUE)
pacman::p_load(here, ggplot2, tidyverse, dplyr)
```

## Importing CSV Files

```{r importing csvs}
piconano = read.csv("Data/tara_phytoplankton_piconano_forhomework_092821.csv")
sample = read.csv ("Data/tara_sample_information_092821.csv" )
```

## Creating Merged, pivoted dataframes

```{r transforming data}
#here we are pivoting the columns containing TARA such that they are data fields.
#made sure that column names are the same between the two data frames so they can be joined!
piconano_long <- piconano %>%
  pivot_longer(
    cols = starts_with("TARA"),
    names_to = "sampleID"
  )
#We are merging them based on the piconano arrangement!
sample_merge <- left_join(piconano_long, sample, by="sampleID")
```

# Homework Questions

## Question 1:

Determine the ten most abundant OTUs, using the column ‘totab.piconano’, which is the sum of all reads of each OTU across all samples.

### Creating top 10 list

```{r top 10 totabs}
top_10_piconano <- piconano %>%
  # Remove duplicate taxogroups, keeping the one with highest abundance
  arrange(desc(totab.piconano)) %>%
  distinct(taxogroup, .keep_all = TRUE) %>%
  # Get top 10
  slice_head(n = 10)

#pivoting this data as well for calculations later
pivoted_top_10_piconano <-  pivot_longer(top_10_piconano,
    cols = starts_with("TARA"),
    names_to = "sampleID"
  )
```

### BLASTING sequences

So based on our results here we can pull out info from the relevant sequence columns and perform a nucleotide BLAST

Plain text will be below but I am also going to create a dataframe of this data from a csv

```{r BLAST df Creation}
list_blast_data = read.csv("Data/top_ten_otus.csv")
```

-   Sequence 1 (cid = 161224):

    -   Most closely related cultured isolate: (Per Ident = 100%) [LC009879.1](https://www.ncbi.nlm.nih.gov/nucleotide/LC009879.1?report=genbank&log$=nuclalign&blast_rank=1&RID=FDCN0KYA014) - Dictyochophyceae

    -   Most closely related uncultured isolate: (Per Ident = 100% ) [JQ781966.1](https://www.ncbi.nlm.nih.gov/nucleotide/JQ781966.1?report=genbank&log$=nuclalign&blast_rank=3&RID=FDDJKS04014) - Stramenopiles

-   Sequence 2 (cid = 171085):

    -   Most closely related cultured isolate: (Per Ident = 94.62%) [MW750340.1](https://www.ncbi.nlm.nih.gov/nucleotide/MW750340.1?report=genbank&log$=nuclalign&blast_rank=1&RID=FDCUHTM0014) - Bacillariophyceae

    -   Most closely related uncultured isolate: (Per Ident = 100%) [KJ757845.1](https://www.ncbi.nlm.nih.gov/nucleotide/KJ757845.1?report=genbank&log$=nuclalign&blast_rank=1&RID=FDCXMG7U015) - uncultured Eukaryote - no info

-   Sequence 3 (cid = 83541):

    -   Most closely related cultured isolate: (Per Ident = 100%) [MT760788.1](https://www.ncbi.nlm.nih.gov/nucleotide/MT760788.1?report=genbank&log$=nuclalign&blast_rank=1&RID=FDD8CWSB014) - Prymnesiophyceae

    -   Most closely related uncultured isolate: (Per Ident = 100%) [KP404791.1c](https://www.ncbi.nlm.nih.gov/nucleotide/KP404791.1?report=genbank&log$=nucltop&blast_rank=15&RID=FDDWYYPE014) - uncultured Eukaryote

-   Sequence 4 (cid = 146146):

    -   Most closely related cultured isolate: (Per Ident = 100%) [AY254857.1](https://www.ncbi.nlm.nih.gov/nucleotide/AY254857.1?report=genbank&log$=nucltop&blast_rank=1&RID=FDE01K2P014) - Dictyochophyceae

    -   Most closely related uncultured isolate: (Per Ident = 100%) [KC583002.1](https://www.ncbi.nlm.nih.gov/nucleotide/KC583002.1?report=genbank&log$=nuclalign&blast_rank=2&RID=FDE2C4EW014) - uncultured Stramenopile

-   Sequence 5 (cid = 73122):

    -   Most closely related cultured isolate: (Per Ident = 90.15) [HG973006.1](https://www.ncbi.nlm.nih.gov/nucleotide/HG973006.1?report=genbank&log$=nucltop&blast_rank=1&RID=FDE5PYPZ014) - Trebouxiophyceae

    -   Most closely related uncultured isolate: (Per Ident = 100%) [KJ760716.1](https://www.ncbi.nlm.nih.gov/nucleotide/KJ760716.1?report=genbank&log$=nucltop&blast_rank=1&RID=FDE87JJB015) - uncultured Eukaryote

-   Sequence 6 (cid = 178099):

    -   Most closely related cultured isolate: (Per Ident = 100%) [MZ611704.1](https://www.ncbi.nlm.nih.gov/nucleotide/MZ611704.1?report=genbank&log$=nucltop&blast_rank=1&RID=FDEACX46014) - Prymnesiophyceae

    -   Most closely related uncultured isolate: (Per Ident = 100%) [HM581637.1](https://www.ncbi.nlm.nih.gov/nucleotide/HM581637.1?report=genbank&log$=nucltop&blast_rank=1&RID=FDED37V3014) - uncultured marine eukaryote

-   Sequence 7 (cid = 231524):

    -   Most closely related cultured isolate: (Per Ident = 100%) [AB058358.1](https://www.ncbi.nlm.nih.gov/nucleotide/AB058358.1?report=genbank&log$=nucltop&blast_rank=1&RID=FDEEZN7V014) - Prymnesiophyceae

    -   Most closely related uncultured isolate: (Per Ident = 100%) [KF62097c1.1](https://www.ncbi.nlm.nih.gov/nucleotide/KF620971.1?report=genbank&log$=nucltop&blast_rank=1&RID=FDEHDNSC014) - Haptophyta

-   Sequence 8 (cid = 78011):

    -   Most closely related cultured isolate: (Per Ident = 100%) [JF791030.1](https://www.ncbi.nlm.nih.gov/nucleotide/JF791030.1?report=genbank&log$=nucltop&blast_rank=1&RID=FDEKBB5T014) - Cryptophyceae

    -   Most closely related uncultured isolate: (Per Ident = 100%) [KF130577.1](https://www.ncbi.nlm.nih.gov/nucleotide/KF130577.1?report=genbank&log$=nucltop&blast_rank=2&RID=FDFS6HAY015) - uncultured eukaryote

-   Sequence 9 (cid = 211092):

    -   Most closely related cultured isolate: (Per Ident = 100%) [KY368637.1](https://www.ncbi.nlm.nih.gov/nucleotide/KY368637.1?report=genbank&log$=nucltop&blast_rank=1&RID=FDFUKB46015) - Mamiellophyceae

    -   Most closely related uncultured isolate: (Per Ident = 100%) [FR874356.1](https://www.ncbi.nlm.nih.gov/nucleotide/FR874356.1?report=genbank&log$=nucltop&blast_rank=1&RID=FDFWRPMB015) - uncultured marine picoeukaryote

-   Sequence 10 (cid = 40687):

    -   Most closely related cultured isolate: (Per Ident = 90.08%) [AB081517.1](https://www.ncbi.nlm.nih.gov/nucleotide/AB081517.1?report=genbank&log$=nucltop&blast_rank=1&RID=FDFYKCCM015) - Dictyochophyceae

    -   Most closely related uncultured isolate: (Per Ident = 90.15%) [DQ647511.1](https://www.ncbi.nlm.nih.gov/nucleotide/DQ647511.1?report=genbank&log$=nucltop&blast_rank=1&RID=FDG5FFRG015) - uncultured marine eukaryote

### Taxonomic Class Comparisons

Compare this information to the ‘lineage’ and ‘taxogroup’ columns in the dataset, which contain the taxonomic annotation performed by the Tara group. How much taxonomic diversity is represented by the top 10 OTUs of piconanoeukaryote phytoplankton (according to these methods)?

```{r comparing taxonomic classes}
taxocomparison <- data.frame(list_blast_data$cult.class, list_blast_data$uncult.class, unique(top_10_piconano$taxogroup), unique(top_10_piconano$lineage))

length(unique(taxocomparison$list_blast_data.cult.class))
length(unique(taxocomparison$list_blast_data.uncult.class))
length(taxocomparison$unique.top_10_piconano.taxogroup)
length(taxocomparison$unique.top_10_piconano.lineage)
```

Based on the above comparisons, I would conclude that the TARA group found more taxonomic diversity based on their methods, than by our top ten analysis.

## Question 2

Let’s look at some variation in distribution among the top ten OTUs. The Tara group took samples at two depths, ‘surface’ (\~3-7 meters, coded SUR) and deep chlorophyll maximum (coded DCM). The column names of the samples in the first file correspond to the Pangaea Accession Number column in the second file, which contains metadata on the samples. Merge the information in these two files in order to classify the OTU samples as surface or DCM. If you do this in Excel, transposing the data and sorting will be helpful. If you do this in R, the function pivot.longer() in the package tidyr, and merge(), may be helpful.

### Merging information

```         
```

### Calculating Relative Abundance

```{r calculating top 10 abundance}
top_10_merge <- left_join(pivoted_top_10_piconano, sample, by = "sampleID")
  
abun_merge <- top_10_merge %>%
  select(value, total.reads, depth, taxogroup) %>%
  mutate(
    RA = value / total.reads,
    rootRA = sqrt(RA)
  )
```

Here I have created a new data frame and added two columns, one with the calculated value and the other is root transformed!

### Making Boxplot

```{r making boxplot}
OTUabund <- ggplot(abun_merge, aes(x=taxogroup, y=sqrt(RA), color=depth))  + geom_boxplot()
OTUabund + theme(
  axis.text.x = element_text(angle = 45, hjust = 1, size = 12)
)
```

## Question 3: Some Hypotheses

**About the Graph**: To generate this figure I first sorted the given list by the "totab.piconano" column, and then returned the top ten "taxogroup" names - I kept the highest totab and filtered out redundants in that column.

**What is the Cause of This Phenomenon?:**

Availability of nutrients and light are going to be the primary driving factors of the phenomena we are observing here.

Some hypothesis for how environmental factors could cause shifts in community composition:

-   I think one of the most consequential environmental factors is most likely nitrogen availability, which would limit an organisms ability to maintain it's metabolism. We know from our lectures and from the incredible data gathered at Station Aloha that nitrogen concentration increases with depth past the DCM.

    -   **Ex Situ:** perhaps using environmental samples we could alter the physical condition of light (light dark bottles in the lab?) and note how diversity and thus nutrient concentrations change over time?

    -   **In Situ:** I think a good in situ experimental approach might be some kind of "Nitrate Ocean Timeseries" (NOT?) where we measure nitrogen concentrations and associate them with microbial diversity.

-   Adding on to the previous hypothesis, how could we demonstrate that *nitrifying archaea* play a key role in the availability of nitrates in the DCM?

    -   ex situ this could be demonstrated by using the aformentioned environmental samples and comparing the production of nitrate under low light conditions with control and archaea added experimental groups - who produces more nitrate?

    -   This could be demonstrated in situ through the use of metagenomics techniques looking for organisms who possess known genes for nitrification, identifying them taxonomically, and comparing nutrient availability - are the archaea there? are they nitrifying? and and is nitrate indeed elevated?
